core/node/state_keeper/src/io/mod.rs

```diff
@@ -137,8 +137,11 @@ pub trait StateKeeperIO: 'static + Send + Sync + fmt::Debug + IoSealCriteria {
 
     /// Blocks for up to `max_wait` until the next transaction is available for execution.
     /// Returns `None` if no transaction became available until the timeout.
-    async fn wait_for_next_tx(&mut self, max_wait: Duration)
-        -> anyhow::Result<Option<Transaction>>;
+    async fn wait_for_next_tx(
+        &mut self,
+        max_wait: Duration,
+        l2_block_timestamp: u64,
```

Comment:
If it's guaranteed that `wait_for_new_batch_params` / `wait_for_new_l2_block_params` output is used for L2 blocks, the L2 block timestamp can be encapsulated inside `MempoolIO`, and this method can be left unchanged.

Reply:
I chose to change the interface because `UpdatesManager` gets initialised (outside)[https://github.com/matter-labs/zksync-era/blob/main/core/node/state_keeper/src/keeper.rs#L151] of the wiring framework. Getting it encapsulated into the `MempoolIO` would require larger refactoring.